extensions/wikia/AdEngine/js/lookup/prebid/adapters/wikia.js

```diff
@@ -0,0 +1,120 @@
+/*global define*/
+define('ext.wikia.adEngine.lookup.prebid.adapters.wikia',[
+	'wikia.document',
+	'wikia.window',
+	'wikia.querystring'
+], function (doc, win, QueryString) {
+	'use strict';
+
+	var bidderName = 'wikia',
+		slots = {
+			oasis: {
+				TOP_LEADERBOARD: {
+					sizes: [
+						[728, 90]
+					]
+				},
+				TOP_RIGHT_BOXAD: {
+					sizes: [
+						[300, 250]
+					]
+				}
+			},
+			mercury: {
+				MOBILE_TOP_LEADERBOARD: {
+					sizes: [
+						[320, 50]
+					]
+				}
+			}
+		},
+		qs = new QueryString();
+
+	function getPrice() {
+		var price = qs.getVal('wikia_adapter', 0);
+
+		return parseInt(price, 10) / 100;
+	}
+
+	function isEnabled() {
+		return getPrice() !== 0;
+	}
+
+	function prepareAdUnit(slotName, config) {
+		return {
+			code: slotName,
+			sizes: config.sizes,
+			bids: [
+				{
+					bidder: bidderName
+				}
+			]
+		};
+	}
+
+	function getSlots(skin) {
+		return slots[skin];
+	}
+
+	function getName() {
+		return bidderName;
+	}
+
+	function getCreative(price, size) {
```

Comment:
maybe extract it to some "creative builder" this is probably should not be a responsibility of this module